src/lib.rs

```diff
@@ -218,105 +258,140 @@ fn encoded_size(bytes_len: usize, config: Config) -> usize {
 ///```
 pub fn encode_config_buf<T: ?Sized + AsRef<[u8]>>(input: &T, config: Config, buf: &mut String) {
     let input_bytes = input.as_ref();
-    let ref charset = match config.char_set {
-        CharacterSet::Standard => tables::STANDARD_ENCODE,
-        CharacterSet::UrlSafe => tables::URL_SAFE_ENCODE,
-    };
 
     // reserve to make sure the memory we'll be writing to with unsafe is allocated
-    buf.reserve(encoded_size(input_bytes.len(), config));
+    let encoded_size = encoded_size(input_bytes.len(), &config);
+    buf.reserve(encoded_size);
 
     let orig_buf_len = buf.len();
-    let mut fast_loop_output_buf_len = orig_buf_len;
 
-    let input_chunk_len = 6;
+    // we're only going to insert valid utf8
+    let mut buf_bytes;
+    unsafe {
+        buf_bytes = buf.as_mut_vec();
+        // expand the string's vec to use its reserved size
+        buf_bytes.set_len(orig_buf_len + encoded_size);
+    }
 
-    let last_fast_index = input_bytes.len().saturating_sub(8);
+    let output_bytes_written = {
+        let mut b64_output = &mut buf_bytes[orig_buf_len..];
 
-    // we're only going to insert valid utf8
-    let mut raw = unsafe { buf.as_mut_vec() };
-    // start at the first free part of the output buf
-    let mut output_ptr = unsafe { raw.as_mut_ptr().offset(orig_buf_len as isize) };
+        // write into the newly reserved space
+        let b64_bytes_written = encode_to_slice(input_bytes, b64_output,
+                                                    config.char_set.encode_table());
+
+        let padding_bytes = if config.pad {
+            add_padding(input_bytes.len(), &mut b64_output[b64_bytes_written..])
+        } else {
+            0
+        };
+
+        let wrappable_bytes = b64_bytes_written + padding_bytes;
+
+        let line_ending_bytes = match config.line_wrap {
+            LineWrap::Wrap(line_len, line_end) =>
+                line_wrap(b64_output, wrappable_bytes, line_len, line_end),
```

Comment:
because we're wrapping only the contents of `b64_output` (which excludes pre-existing contents), #25 is avoided.